packages/mds-compliance-engine/engine/mds-compliance-engine.ts

```diff
@@ -0,0 +1,96 @@
+/*
+    Copyright 2019 City of Los Angeles.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+import { Device, Geography, Policy, UUID, RULE_TYPES } from '@mds-core/mds-types'
+
+import { now, UnsupportedTypeError, uuid } from '@mds-core/mds-utils'
+import { VehicleEventWithTelemetry, ComplianceEngineResult, ComplianceSnapshot } from '../@types'
+import { getProviderIDs, getComplianceInputs, isPolicyActive } from './helpers'
+
+import { processCountPolicy } from './count_processors'
+import { processSpeedPolicy } from './speed_processors'
+import { processTimePolicy } from './time_processors'
+
+function getProcessorType(rule_type: string) {
+  switch (rule_type) {
+    case RULE_TYPES.count: {
+      return processCountPolicy
+    }
+    case RULE_TYPES.speed: {
+      return processSpeedPolicy
+    }
+    case RULE_TYPES.time: {
+      return processTimePolicy
+    }
+    default: {
+      throw new UnsupportedTypeError(`Policy type ${rule_type} unsupported`)
+    }
+  }
+}
+
+export async function createComplianceSnapshot(
+  policy: Policy,
+  provider_id: UUID,
+  geographies: Geography[],
+  processorFunction: (
+    p: Policy,
+    events: VehicleEventWithTelemetry[],
+    geos: Geography[],
+    devicesToCheck: { [d: string]: Device }
+  ) => ComplianceEngineResult | undefined
```

Comment:
May be a good place to use a named type (considering we use this for all of the processors), but not super necessary